packages/tuono/src/dynamic/dynamic.tsx

```diff
@@ -0,0 +1,91 @@
+/**
+ * This component is heavily inspired by Next.js dynamic function
+ * Link: https://github.com/vercel/next.js/blob/1df81bcea62800198884438a2bb27ba14c9d506a/packages/next/src/shared/lib/dynamic.tsx
+ */
+import * as React from 'react'
+
+const isServerSide = typeof window === 'undefined'
+
+interface ComponentModule<T> {
+  default: React.ComponentType<T>
+}
+
+interface DynamicOptions {
+  ssr?: boolean
+  loading?: React.ComponentType<unknown> | null
+}
+
+type Loader<T = object> = () => Promise<
+  React.ComponentType<T> | ComponentModule<T>
+>
+
+interface LoadableOptions<T> extends DynamicOptions {
+  loader: Loader<T>
+}
+
+type LoadableFn = <T = object>(
+  options: LoadableOptions<T>,
+) => React.ComponentType<T>
+
+const defaultLoaderOptions: LoadableOptions<object> = {
+  ssr: true,
+  loading: null,
+  loader: () => Promise.resolve(() => null),
+}
+
+function noSSR<T = object>(
+  LoadableInitializer: LoadableFn,
+  loadableOptions: LoadableOptions<T>,
+): React.ComponentType<T> {
+  if (!isServerSide) {
+    return LoadableInitializer(loadableOptions)
+  }
+
+  if (!loadableOptions.loading) return () => null
+
+  const Loading = loadableOptions.loading
+  // This will only be rendered on the server side
+  function NoSSRLoading(): React.JSX.Element {
+    return <Loading />
+  }
+  return NoSSRLoading
+}
+
+const Loadable = <T = object,>(
+  options: LoadableOptions<T>,
+): React.ComponentType<T> => {
```

Comment:
The code highlight here is broken and also the syntax is look  weird.
Together they make the function hard to read:

<img width="320" alt="image" src="https://github.com/user-attachments/assets/fba31e75-65f6-4bff-b112-c1452a74a8c1" />

I would consider using replacing the arrow function with a regular function:
```suggestion
function Loadable<T = object>(
  options: LoadableOptions<T>,
): React.ComponentType<T> {
```